src/dapla_metadata/datasets/core.py

```diff
@@ -399,6 +423,48 @@ def _extract_metadata_from_existing_document(
             )
             return None
 
+    def _extract_pseudonymization_from_existing_document(
+        self,
+        document: pathlib.Path | CloudPath,
+    ) -> model.PseudonymizationMetadata | None:
+        """Read metadata from an existing metadata document.
+
+        If an existing metadata document is available, this method reads and
+        loads the metadata from it. It validates and upgrades the metadata as
+        necessary. If we have read in a file with an empty "datadoc" structure
+        the process ends.
+        A typical example causing a empty datadoc is a file produced from a
+        pseudonymization process.
+
+        Args:
+            document: A path to the existing metadata document.
+
+        Raises:
+            json.JSONDecodeError: If the metadata document cannot be parsed.
+        """
+        try:
+            with document.open(mode="r", encoding="utf-8") as file:
+                fresh_metadata = json.load(file)
+        except json.JSONDecodeError:
+            logger.warning(
+                "Could not open existing metadata file %s. "
+                "Falling back to collecting data from the dataset",
```

Comment:
```suggestion
                "Could not open existing metadata file %s.",
```
I don't think this is true in this case.